fvtest/threadtest/abortTest.cpp

```diff
@@ -129,64 +129,47 @@ runningMain(void *arg)
 }
 
 typedef struct sleep_testdata_t {
-	omrthread_monitor_t startSync;
-	omrthread_monitor_t exitSync;
-	volatile intptr_t rc;
-	volatile bool started;
+	omrthread_monitor_t sync;
 } sleep_testdata_t;
 
 TEST(ThreadAbortTest, Sleeping)
 {
 	omrthread_t t;
-	sleep_testdata_t mythread;
+	sleep_testdata_t d;
+	sleep_testdata_t *data = &d;
 
-	omrthread_monitor_init(&mythread.startSync, 0);
-	omrthread_monitor_init(&mythread.exitSync, 0);
+	omrthread_monitor_init(&data->sync, 0);
 
-	createDefaultThread(&t, sleepingMain, &mythread);
+	omrthread_monitor_enter(data->sync);
+	createDefaultThread(&t, sleepingMain, data);
+	omrthread_monitor_wait(data->sync); // startup
 
-        mythread.started = false;
-	omrthread_monitor_enter(mythread.startSync);
-	while (!mythread.started) {
-		omrthread_monitor_wait(mythread.startSync);
-	}
-	omrthread_monitor_exit(mythread.startSync);
-
-	omrthread_monitor_enter(mythread.exitSync);
-	omrthread_sleep(START_DELAY);
 	omrthread_abort(t);
```

Comment:
`omrthread_abort(t)` will set `J9THREAD_FLAG_ABORTED` on `t` and interrupt it. Abort will not actually cancel `t` (I.E. stop execution of the physical thread).

`omrthread_sleep_interruptable(timeout, 0)` will return `J9THREAD_PRIORITY_INTERRUPTED` if the thread
has been aborted, either at entry or while sleeping.